DistroLauncher/WSLInfo.cpp

```diff
@@ -0,0 +1,133 @@
+//  License: GPL-3
+//  This program is free software : you can redistribute itand /or modify
+//  it under the terms of the GNU General Public License as published by
+//  the Free Software Foundation, either version 3 of the License.
+//  .
+//  This program is distributed in the hope that it will be useful,
+//  but WITHOUT ANY WARRANTY; without even the implied warranty of
+//  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.See the
+//  GNU General Public License for more details.
+//  .
+//  You should have received a copy of the GNU General Public License
+//  along with this program.If not, see < https://www.gnu.org/licenses/>.
+//  .
+//  On Debian systems, the complete text of the GNU General
+//  Public License version 3 can be found in "/usr/share/common-licenses/GPL-3".
+//
+
+#include "stdafx.h"
+
+namespace Helpers {
+    namespace {
+        bool isX11UnixSocketMounted();
+        bool isWSLVDCPluginRegistered();
+    }
+
+    // Altough currently only used in this translation unit,
+    // this function has a lot of potential to not be exported.
+    bool WslLaunchSuccess(const TCHAR* command, DWORD dwMilisseconds) {
+        DWORD exitCode;
+        HANDLE child;
+        HRESULT hr = g_wslApi.WslLaunch(command, false, NULL, NULL, NULL, &child);
+        if (child == NULL || FAILED(hr)) {
+            return false;
+        }
+
+        WaitForSingleObject(child, dwMilisseconds);
+        auto success = GetExitCodeProcess(child, &exitCode);
+        CloseHandle(child);
+        if (success == FALSE || exitCode != 0) {
+            return false;
+        }
+
+        return true;
+    }
+
+    // Retrieves subsystem version from the WSL API.
+    DWORD WslGetDistroSubsystemVersion() {
+        ULONG distributionVersion;
+        ULONG defaultUID;
+        WSL_DISTRIBUTION_FLAGS wslDistributionFlags;
+        PSTR* defaultEnvironmentVariables;
+        ULONG defaultEnvironmentVariableCount;
+
+        HRESULT hr = g_wslApi.WslGetDistributionConfiguration(&distributionVersion,
+                                                              &defaultUID,
+                                                              &wslDistributionFlags,
+                                                              &defaultEnvironmentVariables,
+                                                              &defaultEnvironmentVariableCount);
+        if (FAILED(hr) || distributionVersion == 0) {
+            Helpers::PrintErrorMessage(hr);
+            return 0;
+        }
+
+        // discards the env variable string array otherwise they will leak.
+        for (int i = defaultEnvironmentVariableCount-1; i >= 0; --i) {
+            CoTaskMemFree((LPVOID)(defaultEnvironmentVariables[i]));
+        }
+
+        return distributionVersion;
+    }
+
+    bool isWslgEnabled() {
+        return isX11UnixSocketMounted() && isWSLVDCPluginRegistered();
+    }
+
+    bool WslGraphicsSupported() {
+        // Could WSL 3 or greater exist in the future?
```

Comment:
good one, I think this is fine this way to assume > 1.

Btw, stupid C++ question, but we can’t compare directly to 1, in this PR and the previous one, I see you always cast it with 1u to unsigned, but the c++ compiler is not smart enough to do the casting at compile time as this is a const?

Reply:
Old habits. 😅 
Compiler is smart enough to figure it out. Technically, that is not a cast, but a way to avoid a cast. Writting the literal as 1u ensures compiler will represent the number 1 appropriatelly as a unsigned integer binary, instead of making it signed and casting later at the moment of comparison. That is not really necessary in simple situations like this one. I'll take it out.

Reply:
I took the opportunity of this readibility check and added a few more thru clang-tidy. There is now a .clang-tidy in DistroLauncher folder with a few checks enabled. Although Visual Studio integrates well with clang-tidy, it will catch a fair amount of warnings from upstream code which I believe we will not address so soon, so instead of making it part of the build system by default, I left it discretely laying on the project folder as an opt-in advisor. It worked quite nicely. I could find a couple of opportunities for improvement in our sources.

Reply:
Excellent! :)